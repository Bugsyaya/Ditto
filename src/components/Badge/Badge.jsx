import React from 'react';
import { Badge as AntdBadge } from 'antd'

const Badge = ({
  props,
  color,
  count,
  dot,
  offset,
  overflowCount,
  showZero,
  size,
  status,
  text,
  title
}) => {
  return <AntdBadge
    color={color}
    count={count}
    dot={dot}
    offset={offset}
    overflowCount={overflowCount}
    showZero={showZero}
    size={size}
    status={status}
    text={text}
    title={title}
    {...props} />
}

export { Badge };