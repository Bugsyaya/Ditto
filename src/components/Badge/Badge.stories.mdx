import { Meta, ArgsTable } from '@storybook/addon-docs/blocks';
import {Badge} from './Badge';


<Meta
  title="Components/Badge"
  argTypes={{
    color: {
      name:'color', 
      description:'color', 
      control: {
        type: 'color'
      },
    },
    count: {
      description:'count', 
      control: {
        type: 'number'
      }
    },
    dot: {
      description:'dot', 
      control: {
        type: 'boolean'
      },
      table: {
        defaultValue: {
          summary: 'false',
        }
      }
    },
    offset: {
      description:'offset', 
      control: {
        type: '[number, number]',
      }
    },  
    overflowCount: {
      description: 'overflowCount',
      control: {
        type: 'number'
      },
      table: {
        defaultValue: {
          summary: '99'
        }
      }
    },
    showZero: {
      description:'showZero', 
      control: {
        type: 'boolean'
      },
      table: {
        defaultValue: {
          summary: 'false'
        }
      }
    },
    size: {
      description:'icon', 
      control: {
        type: 'select',
        options: [
          'default',
          'small'
        ]
      },
      table: {
        defaultValue: {
          summary: 'default'
        }
      }
    },
    status: {
      description:'loading', 
      control: {
        type: 'select',
        options: [
          'success',
          'processing',
          'default',
          'error',
          'warning'
        ]
      },
      table: {
        defaultValue: {
          summary: 'default'
        }
      }
    },
    text: {
      description:'text', 
      control: {
        type: 'text',
      }
    },
    title: {
      description:'title', 
      control: {
        type: 'text'
      }
    }
  }}
  args={{
    count: 12,
  }}
/>

### Button

Use this playground to test the button component

<Canvas>
  <Story name="Standard">
    {args => {
      return <Badge {...args} />;
    }}
  </Story>
</Canvas>


<ArgsTable story="Standard" />