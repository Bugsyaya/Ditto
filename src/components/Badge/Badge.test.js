import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import React from 'react';
import {Badge} from './Badge';

describe('Badge component', () => {
  const textContent = 'LOGIN';

  const badge = (
    <Badge >{textContent}</Badge>
  );

  it('Snapshot button', () => {
    const { baseElement } = render(badge);
    expect(baseElement).toMatchSnapshot();
  });
});