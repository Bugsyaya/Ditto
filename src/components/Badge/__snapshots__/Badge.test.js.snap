// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component Snapshot button 1`] = `
<body>
  <div>
    <span
      class="ant-badge ant-badge-not-a-wrapper"
    />
  </div>
</body>
`;
